File: frontend/bff/src/application/counterparties/dtos/counterparty.dto.ts
import type { CounterpartyType } from '../../../cross-cutting/enums/counterparty-type.js';

export interface CounterpartyDto {
  id: string;
  /** Código legível da organização (CP-01). */
  code: string;
  name: string;
  type: CounterpartyType;
  document: string | null;
  address: string | null;
  country: string | null;
  isHomologated: boolean;
  notionalLimitUsd: number | null;
  mtmLimitUsd: number | null;
}

File: backend/core-service/src/Fix.Application/UseCases/Counterparties/Dtos/CounterpartyDto.cs
using Fix.Domain.AggregateRoots.Counterparties;

namespace Fix.Application.Counterparties.Dtos;

public sealed record CounterpartyDto(
    Guid Id,
    string Code,
    string Name,
    CounterpartyType Type,
    string? Document,
    string? Address,
    string? Country,
    bool IsHomologated,
    decimal? NotionalLimitUsd,
    decimal? MtmLimitUsd);

